### PR TITLE
Define build tool version requirements for consistent dev/build environments

### LINKED ISSUE #209: Define build tool version requirements for consistent dev/build environments
### Describe the bug
Issues can occur when building or developing with an untested version of Composer or Node.
To help us achieve consistent development and build environments, we should **define and document** build tool version requirements.

I suggest that we use LTS versions (rather than latest/stable) of both Node and Composer to ensure stability and infrequent churn.

### Proposed steps to define build tool versions
* Add a **Development** section to the repo’s  `README.md` which lists pre-requisite tools and the intention to use LTS versions.
	* Node: LTS
	* Composer: LTS
	* [nvm](https://github.com/nvm-sh/nvm#nvmrc) (optional and recommended for using the required node version)
* Include the required **Node** version in:
   * `.nvmrc` (`lts/*`)
   * [`package.json/engines`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines) (explicit version number of Node and NPM)
* Test and update defined versions with each new LTS version. This can happen ad-hoc, e.g. when a team member notices that a new LTS version has been released.

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #209 

## Description

This PR:
* Adds a **Prerequisites** section to the repo’s  `README.md` which lists pre-requisite tools and the intention to use LTS versions.
	* Node: LTS (currently v16.17.1)
	* Composer: LTS (currently v2.2.18)
	* [nvm](https://github.com/nvm-sh/nvm#nvmrc) (optional and recommended for using the required node version)
* Defines the required **Node** version in:
   * `.nvmrc` (`lts/*`)
   * [`package.json/engines`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines) (node v16.17.1 npm v8.15.0)

## How to test this PR

1. Install [nvm](https://github.com/nvm-sh/nvm#nvmrc) if you haven't already
2. Run `nvm install`. The current LTS version of node/npm should be installed (node v16.17.1 npm v8.15.0)
3. Run `npm install`. No errors should occur.
4. [Download and install the LTS version of Composer](https://getcomposer.org/download/) as defined in README.md on this branch.
5. Run `composer install`
6. Run `npm run build`. The build should occur with no errors.
7. Run `npm run build` a second time.
8. Examine the contents of `woocommerce-subscriptions-core.zip`. 
   - `node_modules` should not be included.
   - `woocommerce-subscriptions-core.zip` should not be included.
9. Upload & install `woocommerce-subscriptions-core.zip` to a test WP site (e.g. JN) alongside WC Subs or WCPay and ensure that plugin can be activated without errors.
 
## Product impact

- [x] Added changelog entry (or does not apply)
- [x] Will this PR affect WooCommerce Subscriptions? NO
- [x] Will this PR affect WooCommerce Payments? NO

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,9 +38,15 @@ new WC_Subscriptions_Core_Plugin();
 
 This repository can also be loaded as standalone plugin, for development purposes only.
 
+### Prerequisites
```
Comment (Jinksi) [inline README.md:41]: This wording is inspired by the [WC README Prerequisites](https://github.com/woocommerce/woocommerce/blob/trunk/README.md#prerequisites)

File: README.md [hunk 2]
```diff
@@ -38,9 +38,15 @@ new WC_Subscriptions_Core_Plugin();
...
+- [Node](https://nodejs.org/en/) LTS v16.x.x
+- [Composer](https://getcomposer.org/doc/00-intro.md) LTS v2.2.x
+- [NVM](https://github.com/nvm-sh/nvm#installing-and-updating): We recommend using NVM to ensure you’re aligned with the expected Node version. This repository contains [an `.nvmrc` file](.nvmrc) which helps ensure you are using the correct version of Node.
+
 After cloning the repo, install dependencies and build:
 
 ```
+nvm use
```
Comment (brucealdridge) [inline README.md:49]: not part of your commit, but can you remove `&& composer install` as that is a step included in `npm install`

File: package.json [hunk 3]
```diff
@@ -42,8 +42,8 @@
 		"postarchive": "unzip woocommerce-subscriptions-core.zip -d woocommerce-subscriptions-core && rm woocommerce-subscriptions-core.zip && zip -r woocommerce-subscriptions-core.zip woocommerce-subscriptions-core && rm -rf woocommerce-subscriptions-core"
 	},
 	"engines": {
-		"node": ">=0.8.0",
-		"npm": ">=1.1.0"
+		"node": "^16.17.1",
+		"npm": "^8.15.0"
```
Comment (brucealdridge) [inline package.json:46]: I thought this would limit usage/install to these versions, but it doesn't. TIL.

**Review by brucealdridge — APPROVED** [review]
Tested w/ `nvm use && npm install`
LGTM.

File: README.md (hunk 2 above)
Comment (Jinksi) [reply README.md:49]: Thanks! updated in 46da4a0

**Pull request merged**